pkg/apps/gitops.go

```diff
@@ -166,9 +176,18 @@ func (o *GitOpsOptions) DeleteApp(app string, alias string, autoMerge bool) erro
 
 // GetApps retrieves all the apps information for the given appNames from the repository and / or the CRD API
 func (o *GitOpsOptions) GetApps(appNames map[string]bool, expandFn func([]string) (*v1.AppList, error)) (*v1.AppList, error) {
-	dir, _, _, _, err := gits.ForkAndPullRepo(o.DevEnv.Spec.Source.URL, o.EnvironmentsDir, o.DevEnv.Spec.Source.Ref, "master", o.GitProvider, o.Gitter, "")
```

Comment:
As much as I also dislike the use of `ForAndPullRepo` here, if for no other reason than the fact that it is a terrible name, I believe this would actually work, if there were not the #5772 bug.

Reply:
Another issue with ~/.jx/environment is that in the case of app it is not keys against context. So if you have multiple clusters and you switch between them you also get an error, this time a legit merge error. This is due to the fact that this app code came first. Later code which uses JX_HOME to cache checkout is keying against the cluster. I believe there is an issue for this as well.

Reply:
But _why_ use `ForkAndPullRepo` here? It's the forking that drove me off - there's absolutely no reason to _fork_ the dev env as part of running `jx get apps`.

Reply:
Exactly, the name is terrible. It implies completely different semantics to what is actually happening, depending on the passed parameters. It is not always forking. The docs also says "ForkAndPullRepo pulls the specified gitUrl into dir using gitter, creating a remote fork if needed using the git provider". But that makes the function not less confusing. It is one of these cases where a single methods tries to do everything. 

I agree it completely confuses the reader of the code at this point in the code. I also think there are several things wrong with ForkAndPullRepo, so the less we use it the better. 

Doing a fresh throw-away checkout for each of these app operations will fix it, but with a pretty high cost. It feels like it there should be a better way. But let's go for something which works.

Reply:
We already effectively are doing throw away checkouts for `jx add app`, `jx delete app`, `jx upgrade app`, and their friends (namely `jx step scheduler [apply|migrate]`. I just threw in a throwaway checkout for `jx get apps` too. =)